llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -343,6 +343,52 @@ RISCVTTIImpl::getConstantPoolLoadCost(Type *Ty,  TTI::TargetCostKind CostKind) {
                              /*AddressSpace=*/0, CostKind);
 }
 
+InstructionCost
+RISCVTTIImpl::isMultipleInsertSubvector(VectorType *Tp, ArrayRef<int> Mask,
+                                        TTI::TargetCostKind CostKind) {
```

Comment:
Can we split this function into something similar to `ShuffleVectorInst::isReplicationMask` and the other mask functions, e.g. `static bool isRepeatedConcatMask(ArrayRef<int> Mask, int &SubvectorSize)`.

Then call that check in getShuffleCost and compute the cost there if it's true. That way it's a bit more similar to how we check other mask patterns.

Reply:
Are you saying that we should add this into `ShuffleVectorInst` and let `RISCVTTIImpl::getShuffleCost` call it?

Reply:
Not move it into getShuffleCost, but just convert it into a static function that only checks the mask

Reply:
please check https://github.com/llvm/llvm-project/pull/111459/commits/39c1368f9cfa6540da97bc2f2bc4f3d60b5e7ac2